lib/api/include/irods/ticketAPI.hpp

```diff
@@ -0,0 +1,414 @@
+#ifndef IRODS_TICKET_API_HPP
+#define IRODS_TICKET_API_HPP
+
+#include <string_view>
+#include <type_traits>
+#include <iostream>
+#include <string>
+
+#undef NAMESPACE_IMPL
+#undef RxComm
+#undef rxTicketAdmin
+
+#ifdef IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define NAMESPACE_IMPL server
+#    define RxComm         RsComm
+#    define rxTicketAdmin  rsTicketAdmin
+
+struct RsComm;
+#else
+#    define NAMESPACE_IMPL client
+#    define RxComm         RcComm
+#    define rxTicketAdmin  rcTicketAdmin
+
+struct RcComm;
+#endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+/**
+ * @brief Namespace to contain the API for tickets
+ *
+ */
+namespace irods::administration::ticket
+{
+    inline struct admin_tag
+    {
+    } admin;
+
+    enum class ticket_type
+    {
+        read,
+        write
+    };
+
+    enum class ticket_property
```

Comment:
In usage, these types will be like this:
```
irods::administration::ticket::ticket_type
irods::administration::ticket::ticket_property
```
How do we feel about this?
```
irods::administration::ticket::type
irods::administration::ticket::property
```
Too terse?

Reply:
I feel it is a little too terse, but I'm willing to be convinced.

Reply:
I think this may the way to do it. It seems a little redundant because the ticket namespace is already specified.

Reply:
Yes, it is redundant, but if we ever introduce more types that want to use `property` in the name, then now we have a problem.

Reply:
Users of the API can always use type aliases to introduce shorter names.

Reply:
Based on further clarification, type and property are quite generic. Sticking with ticket_type and ticket_property are probably way to go